kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/ResourceDefinitionContext.java

```diff
@@ -45,6 +45,15 @@ public String getKind() {
   public boolean isNamespaceScoped() {
     return namespaced;
   }
+  
+  protected void validate() {
+    if (plural == null) {
+      if (kind == null) {
+        throw new IllegalArgumentException("Neither kind nor plural was set, at least one is required");
+      }
+      plural = Utils.getPluralFromKind(kind);
```

Comment:
This part is tricky because the target class might use a different plural version than the one computed here.

Reply:
If the context is constructed from a class it will use the annotation to populate this field.  This will only occur in case where someone is manually constructing the context.  Every method the specifies both a context and a class has been deprecated - it's not expected that a user will manually construct the context when a class is used.